numba/npyufunc/omppool.cpp

```diff
@@ -90,6 +131,10 @@ parallel_for(void *fn, char **args, size_t *dimensions, size_t *steps, void *dat
     // index variable in OpenMP 'for' statement must have signed integral type for MSVC
     const ptrdiff_t size = (ptrdiff_t)dimensions[0];
 
+    // holds the shared variable for `num_threads`, this is a bit superfluous
+    // but present to force thinking about the scope of validity
+    int agreed_nthreads = num_threads;
```

Comment:
Is this still needed? My reading of the code is that `agreed_nthreads` and `num_threads` should always be equal. Would some kind of check / assert that all threads agree on the number of threads provide a stronger guarantee of what's being reasoned about here?

Reply:
Or is it that "superfluous" is not about how long this code needs to exist, but it's expressing that the validity of `agreed_nthreads` is only transient, perhaps between its definition here and its use on line 164?

Reply:
The superfluousness here is that as per your first comment, they are the same thing... but this is to force thinking about scopes. The main thread knows at this point what `num_threads` is, by copying this into another variable it's with the intention of being explicit about there being a shared variable which holds the "agreed" thread mask and that this is used to broadcast it to the other threads in the team so that all the threads see the same value.